selftests/unit/test_remote.py

```diff
@@ -25,154 +30,80 @@ class RemoteTestRunnerTest(unittest.TestCase):
-                        remote_username='username',
-                        remote_hostname='hostname',
-                        remote_port=22,
-                        remote_password='password',
-                        remote_key_file=None,
-                        remote_timeout=60,
-                        show_job_log=False,
-                        env_keep=None)
-        log = flexmock()
-        log.should_receive("info")
-        job = flexmock(args=Args, log=log)
-        self.runner = avocado_runner_remote.RemoteTestRunner(job, None)
-
-    def tearDown(self):
-        flexmock_teardown()
-
-    def test_setup(self):
-        """ Tests RemoteResult.test_setup() """
-        self.runner.setup()
-        flexmock_teardown()
+        """
+        Test RemoteTestRunner.run_suite()
+
+        The general idea of this test is to:
+
+        1) Create the machinery necessary to get a RemoteTestRunner
+           setup inside a job, or looking at it the other way around, to
+           have a runner that is created with a valid job.
+
+        2) Mock the interactions with a remote host.  This is done here
+           basically by mocking 'Remote' and 'fabric' usage.
+
+        3) Provide a polluted JSON to be parsed by the RemoteTestRunner
+
+        4) Assert that those results are properly parsed into the
+           job's result
+        """
+        job_args = argparse.Namespace(test_result_total=1,
+                                      remote_username='username',
+                                      remote_hostname='hostname',
+                                      remote_port=22,
+                                      remote_password='password',
+                                      remote_key_file=None,
+                                      remote_timeout=60,
+                                      show_job_log=False,
+                                      mux_yaml=['~/avocado/tests/foo.yaml',
+                                                '~/avocado/tests/bar/baz.yaml'],
+                                      dry_run=True,
+                                      env_keep=None,
+                                      reference=['/tests/sleeptest.py',
+                                                 '/tests/other/test',
+                                                 'passtest.py'])
+
+        try:
+            job = Job(job_args)
+            runner = avocado_runner_remote.RemoteTestRunner(job, job.result)
+            return_value = (True, (version.MAJOR, version.MINOR))
+            runner.check_remote_avocado = mock.Mock(return_value=return_value)
+
+            # These are mocked at their source, and will prevent fabric from
+            # trying to contact remote hosts
+            with mock.patch('avocado_runner_remote.Remote'):
+                with mock.patch('avocado_runner_remote.fabric'):
+                    runner.remote = avocado_runner_remote.Remote(job_args.remote_hostname)
```

Comment:
It's interesting that this works, because the `runner.remote` should be replaced in `run_suite` in `setup`. Anyway I verified it works well and returns the expected value.